src/Controller/User/UserParticipationController.php

```diff
@@ -0,0 +1,153 @@
+use App\UX\UserChartBuilder;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Workflow\StateMachine;
+
+class UserParticipationController extends AbstractController
+{
+    public function __construct(
+        private ParticipationRepository $participationRepository,
+        private ConferenceRepository $conferenceRepository,
+        private UserChartBuilder $userChartBuilder,
+        private EntityManagerInterface $em,
+        private StateMachine $workflow,
+    ) {
+    }
+
+    #[Route(path: '/user/participations', name: 'user_participations')]
+    public function userParticipation(Request $request): Response
+    {
+        $participation = new Participation($this->getUser());
+        $form = $this->createForm(ParticipationType::class, $participation);
+
+        if ($form->handleRequest($request)->isSubmitted() && $form->isValid()) {
+            $this->em->persist($participation);
+            $this->em->flush();
+
+            $this->addFlash('info', 'Participation Submitted');
+
+            return $this->redirectToRoute('user_participations');
+        }
+
+        $futureParticipations = $this->participationRepository->findFutureParticipationsByUser($this->getUser());
+        $pendingParticipations = $this->participationRepository->findPendingParticipationsByUser($this->getUser());
+        $rejectedParticipations = $this->participationRepository->findRejectedParticipationsByUser($this->getUser());
+        $pastParticipations = $this->participationRepository->findPastParticipationsByUser($this->getUser());
+
+        if (0 !== \count($pendingParticipations) + \count($futureParticipations) + \count($rejectedParticipations) + \count($pastParticipations)) {
+            $chart = $this->userChartBuilder->buildParticipationsChart($pendingParticipations, $futureParticipations, $rejectedParticipations, $pastParticipations);
+        }
+
+        return $this->render('user/participation/participation.html.twig', [
+            'form' => $form->createView(),
+            'futureParticipations' => $futureParticipations,
+            'pendingParticipations' => $pendingParticipations,
+            'rejectedParticipations' => $rejectedParticipations,
+            'pastParticipations' => $pastParticipations,
+            'chart' => $chart ?? null,
+        ]);
+    }
+
+    #[Route(path: '/user/participation-cancel/{id}', name: 'user_participations_cancel')]
+    public function cancelParticipation(Participation $participation): Response
+    {
+        if ($this->getUser() !== $participation->getParticipant()) {
+            throw new AccessDeniedException('You are not allowed to cancel someone else\'s participation.');
+        }
```

Comment:
It should be handler by the security component, with a voter.

So I expect something like:

```
#IsGranted('PARTICIPATION_CANCEL', participation)
```
or in plain PHP, it's up to you

Because ATM, you are duplicating the logic everywhere you need it.
With a voter, the logic is only in the voter (and could be tested easilly)